File: VB/ChartsDemo.Wpf/ViewModels/RegionPopulationData.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Xml.Serialization

Namespace ChartsDemo
    <XmlRoot("RegionPopulationData")>
    Public Class RegionPopulationData
        Inherits List(Of RegionPopulation)


        Private Shared data_Renamed As List(Of RegionPopulation) = Nothing

        Public Shared ReadOnly Property Data() As List(Of RegionPopulation)
            Get
                If data_Renamed Is Nothing Then
                    Dim s = New XmlSerializer(GetType(RegionPopulationData))
                    data_Renamed = DirectCast(s.Deserialize(DataLoader.LoadFromResources("/Data/RegionPopulationData.xml")), List(Of RegionPopulation))
                End If
                Return data_Renamed
            End Get
        End Property
    End Class
    Public Class RegionPopulation
        Public Property Region() As String
        Public Property Items() As List(Of PopulationByYear)
    End Class
    Public Class PopulationByYear
        Public Property Year() As Integer
        Public Property Population() As Integer
    End Class
End Namespace

File: CS/ChartsDemo.Wpf/ViewModels/RegionPopulationData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;

namespace ChartsDemo {
    [XmlRoot("RegionPopulationData")]
    public class RegionPopulationData : List<RegionPopulation> {
        static List<RegionPopulation> data = null;

        public static List<RegionPopulation> Data {
            get {
                if(data == null) {
                    var s = new XmlSerializer(typeof(RegionPopulationData));
                    data = (List<RegionPopulation>)s.Deserialize(DataLoader.LoadFromResources("/Data/RegionPopulationData.xml"));
                }
                return data;
            }
        }
    }
    public class RegionPopulation {
        public string Region { get; set; }
        public List<PopulationByYear> Items { get; set; }
    }
    public class PopulationByYear {
        public int Year { get; set; }
        public int Population { get; set; }
    }
}
